tools/src/test/java/org/apache/kafka/tools/JmxToolTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.common.utils.Exit;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+
+import java.lang.management.ManagementFactory;
+import java.net.ServerSocket;
+import java.rmi.registry.LocateRegistry;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class JmxToolTest {
+    private final ToolsTestUtils.MockExitProcedure exitProcedure = new ToolsTestUtils.MockExitProcedure();
+
+    private static JMXConnectorServer jmxAgent;
+    private static String jmxUrl;
+
+    @BeforeAll
+    public static void beforeAll() throws Exception {
+        int port = findRandomOpenPortOnAllLocalInterfaces();
+        jmxAgent = startJmxAgent(port);
+        jmxUrl = String.format("service:jmx:rmi:///jndi/rmi://:%d/jmxrmi", port);
+    }
+
+    @AfterAll
+    public static void afterAll() throws Exception {
+        jmxAgent.stop();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        Exit.setExitProcedure(exitProcedure);
+    }
+
+    @AfterEach
+    public void afterEach() {
+        Exit.resetExitProcedure();
+    }
+
+    @Test
+    public void kafkaVersion() {
+        String out = executeAndGetOut("--version");
+        assertNormalExit();
+        assertEquals(AppInfoParser.getVersion(), out);
```

Comment:
This fails when running in Intellij. The output is:
```
[2023-02-01 16:49:33,671] WARN Error while loading kafka-version.properties: inStream parameter is null (org.apache.kafka.common.utils.AppInfoParser:46)
unknown
```
I guess all we could simply do `assertTrue(out.contains(AppInfoParser.getVersion()));`

Reply:
Looks like it's working on my end.
![Screenshot from 2023-02-01 17-15-54](https://user-images.githubusercontent.com/11456498/216100017-c5f290c6-f964-41dd-8d79-d5cf0a27b379.png)